generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpecFactory.java

```diff
@@ -11,15 +12,24 @@
 
 import java.math.BigDecimal;
 import java.time.LocalDateTime;
-import java.util.Collection;
+import java.util.*;
 import java.util.HashMap;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Map;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 public class FieldSpecFactory {
+    private final FieldSpecMerger fieldSpecMerger;
+
+    @Inject
+    public FieldSpecFactory(FieldSpecMerger fieldSpecMerger) {
+        this.fieldSpecMerger = fieldSpecMerger;
+    }
+
+    public FieldSpecFactory() {
```

Comment:
if this constructor has been added for the tests, then it should be removed. we can inject the FieldSpecMerger during tests

Reply:
good spot, will remove that constructor and add `new FieldSpecMerger()` to the required tests